tests/simulator/test_advance.py

```diff
@@ -237,6 +240,106 @@ def test_field_coarsening_via_subcycles(self, refinement_boxes):
+              if it had no refined level to the level field ghost of L1 of L0L1_datahier
+        """
+
+        print("test_field_coarsening_via_subcycles for dim/interp : {}/{}".format(ndim, interp_order))
+
+        from tests.amr.data.field.refine.test_refine_field import refine
+        from tests.core.numerics.interpolator.interpolator_test import interpolate_datasets
+        from pyphare.pharein import global_vars
+
+        time_step_nbr = 1 # after first advance, simulations diverge
+
+        def _getHier(diag_dir, boxes=[]):
+            return self.getHierarchy(interp_order, boxes, "eb", cells=30,
+                time_step_nbr=time_step_nbr, smallest_patch_size=5, largest_patch_size=30,
+                diag_outputs=diag_dir, extra_diag_options={"fine_dump_lvl_max": 10}, time_step=0.001,
+            )
+
+        L0_datahier = _getHier(f"phare_lvl_ghost_interpolation_L0_diags_{self.ddt_test_id()}")
+        L0L1_datahier = _getHier(
+          f"phare_lvl_ghost_interpolation_L0L1_diags_{self.ddt_test_id()}", refinement_boxes
+        )
+
+        lvlSteps = global_vars.sim.level_time_steps
+        assert len(lvlSteps) == 2  # this test is only configured for L0 -> L1 refinement comparisons
+
+        def assert_time_in_hier(*ts):
+            for t in ts:
+                print(f"checking time {t} exists")
+                self.assertIn(L0L1_datahier.format_timestamp(t), L0L1_datahier.times())
+
+        coarsest_time_before = 0 # init
+        coarsest_time_after = coarsest_time_before + lvlSteps[0]
+        assert_time_in_hier(coarsest_time_before, coarsest_time_after)
+
+        coarseBeforePatches = L0_datahier.level(0, coarsest_time_before).patches
+        coarseAfterPatches = L0_datahier.level(0, coarsest_time_after).patches
+        assert len(coarseBeforePatches) == len(coarseAfterPatches)
+
+        for coarsePatch_idx in range(len(coarseBeforePatches)):
+            coarseBeforePatch = coarseBeforePatches[coarsePatch_idx]
+            coarseAfterPatch  = coarseAfterPatches[coarsePatch_idx]
+            assert coarseBeforePatch.box == coarseAfterPatch.box
+
+            for finer_subcycle in range(0, global_vars.sim.level_step_nbr[1] + 1):
+
+                finer_subcycle_time = coarsest_time_before + (lvlSteps[1] * finer_subcycle)
+                assert_time_in_hier(finer_subcycle_time)
+                finer_subcycle_patches = L0L1_datahier.level(1, finer_subcycle_time).patches
+
+                for finer_subcycle_patch in finer_subcycle_patches:
+                    if boxm.coarsen(finer_subcycle_patch.box, refinement_ratio) in coarseBeforePatch.box:
+                        for EM in ["E", "B"]:
+                            for xyz in ["x", "y", "z"]:
+                                qty = f"{EM}{xyz}"
+                                assert qty in coarseBeforePatch.patch_datas
+
+                                coarseBefore_pd = coarseBeforePatch.patch_datas[qty]
```

Comment:
you can also define coarseAfter_pd, to simplify the call to interpolate_datasets later

Reply:
ok